src/components/widgets/PAutoComplete.js

```diff
@@ -1,13 +1,48 @@
-import { PInput, PListItem, PList } from '../elements'
 import React from 'react'
+import { PInput, PListItem, PList } from '../elements'
+import styles from './PAutoComplete.module.css'
+
+function escapeRegExp(str) {
+  return str.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, '\\$&')
+}
+
+const getHighlightedText = (text, keywork) => {
+  const escapedkeywork = escapeRegExp(keywork)
+  const parts = text.split(new RegExp(`(${escapedkeywork})`, 'gi'))
```

Comment:
Be careful when using `RegExp`, there will be some bugs regarding to special character and non-unicode chars.

Reply:
Hi @longtran-pops , so far I think cache is a better way to improve searching time and I have updated the source code. If it's the right solution I will think more about the caching strategy (for now it's quite simple)